### PR TITLE
Support `node:` prefix for more modern node

### PR DESCRIPTION
**Pull request by demosjarco**
Some edge hosting services now support runtime node polyfills, but they require the `node:` prefix that is supported as of v12.20.0, v14.13.1, v16 and later

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# To get started with Dependabot version updates, you'll need to specify which
```
Comment (fent) [inline .github/dependabot.yml:1]: please separate into a separate PR

File: .github/workflows/nodejs.yml [hunk 2]
```diff
@@ -7,24 +7,27 @@ jobs:
...
         npm run build --if-present
         npm test
 
-    - uses: codecov/codecov-action@v1.0.3
-      with:
-        token: ${{ secrets.CODECOV_TOKEN }}
+    # Disable codecov for now
+    # - uses: codecov/codecov-action@v1.0.3
+    #   with:
+    #     token: ${{ secrets.CODECOV_TOKEN }}
```
Comment (fent) [inline .github/workflows/nodejs.yml:33]: why disable it?

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+name: Publish Package to npmjs
```
Comment (fent) [inline .github/workflows/publish.yml:1]: place in separate PR

File: package-lock.json [hunk 4]
```diff
@@ -1,24 +1,46 @@
 {
-  "name": "feedme",
+  "name": "feedme-modern",
```
Comment (fent) [inline package-lock.json:2]: revert?

File: package.json [hunk 5]
```diff
@@ -34,15 +36,15 @@
   "devDependencies": {
     "@types/clarinet": "^0.12.0",
     "@types/mocha": "^8.0.4",
-    "@types/node": "^14.14.10",
+    "@types/node": "^16.18.32",
```
Comment (fent) [inline package.json:39]: i'm not sure this needs to be updated

File: package.json [hunk 6]
```diff
@@ -25,7 +26,8 @@
     "prepare": "tsc -p tsconfig.build.json",
     "build": "tsc -p tsconfig.build.json",
     "test": "nyc --extension .ts --reporter=lcov --reporter=text-summary npm run test:unit",
-    "test:unit": "mocha --require ts-node/register test/*-test.ts"
+    "test:unit": "mocha --require ts-node/register test/*-test.ts",
+    "prepublishOnly": "npm run build"
```
Comment (fent) [inline package.json:30]: `prepare` already runs before publish

File: package.json [hunk 7]
```diff
@@ -34,15 +36,15 @@
...
+    "@types/node": "^16.18.32",
     "@types/sax": "^1.2.1",
     "mocha": "^6.0.0",
     "nyc": "^15.1.0",
     "ts-node": "^9.0.0",
     "typescript": "^4.1.2"
   },
   "engines": {
-    "node": ">=10"
+    "node": ">=16.0.0"
```
Comment (fent) [inline package.json:47]: this page says it was introduced in v14 https://nodejs.org/api/modules.html#core-modules

File: test/atom-test.ts [hunk 8]
```diff
@@ -1,5 +1,5 @@
 import FeedMe from '..';
-import fs     from 'fs';
+import fs     from 'node:fs';
 import path   from 'path';
 import assert from 'assert';
```
Comment (fent) [inline test/atom-test.ts:4]: why not change `path` and `assert` modules too? they may not be to be polyfilled, but we should be consistent with the way core modules are imported

File: package.json [hunk 9]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "feedme",
+  "name": "feedme-modern",
```
Comment (fent) [inline package.json:2]: revert

File: package.json [hunk 10]
```diff
@@ -11,11 +11,12 @@
   "version": "2.0.2",
   "repository": {
     "type": "git",
-    "url": "git://github.com/fent/feedme.js.git"
+    "url": "git://github.com/demosjarco/feedme.js.git"
   },
-  "author": "fent <fentbox@gmail.com> (https://github.com/fent)",
+  "author": "DemosJarco <1209494+demosjarco@users.noreply.github.com> (https://github.com/demosjarco)",
```
Comment (fent) [inline package.json:16]: revert. you can add yourself to `contributors`

**Review by fent — CHANGES_REQUESTED** [review]
Hi, thank you for taking the time to contribute

I'm curious what the use case is here, are you trying to run this in the browser, and the polyfills would only work with `node:`? should i be using this new syntax going forward and on other node modules i published? 😳

please split up the PR into changes related to its main intention. if you'd like to also submit dependabot changes for example, do so in a separate PR. it's better to keep changes small so that the history of this library is more easily digestible by others looking through it.

File: package.json [hunk 11]
```diff
@@ -8,14 +8,15 @@
     "json",
     "parser"
   ],
-  "version": "2.0.2",
+  "version": "2.1.0",
```
Comment (fent) [inline package.json:11]: updating the version is handled by the release. and it would be a major release since the PR would drop support for older node versions

**Comment by demosjarco** [comment]
Oh sorry, I thought I was committing to my branch but I was committing to this PR. This will be a mess to clean up so I'll just reopen a new PR. As for the the node prefix, I'm trying to find their blog post mentioning it, but all I could find right now is

> The node: prefix denotes the loading of a core module

[https://nodejs.org/en/blog/announcements/v18-release-announce#test-runner-module-experimental](https://nodejs.org/en/blog/announcements/v18-release-announce#test-runner-module-experimental)

The environment I'm trying to use is [Cloudflare Workers](https://developers.cloudflare.com/workers/runtime-apis/nodejs) but they require the `node:` prefix or else it won't load them.
